### PR TITLE
Add GTI to Dataset

### PR DESCRIPTION
**Pull request by JouvinLea**
Start to implement the GTI in the Dataset Class.

### CODE REVIEW

File: gammapy/spectrum/dataset.py [hunk 1]
```diff
@@ -805,6 +813,12 @@ def stack_edisp(self):
         irf_stacker.stack_edisp()
         self.stacked_edisp = irf_stacker.stacked_edisp
 
+    def stack_gti(self):
+        """Stack GTI
+        """
+        # TODO: use what Regis is implemented for GTI union
+        self.stack_gti = None
```
Comment (registerrier) [inline gammapy/spectrum/dataset.py]: you should have typed `self.stacked_gti = None`.
Anyway I would `raise NotImplemented`

File: gammapy/spectrum/extract.py [hunk 2]
```diff
@@ -134,6 +134,7 @@ def process(self, observation, bkg):
             backscale=1,
             backscale_off=bkg.a_off,
             obs_id=observation.obs_id,
+            gti=observation.gti,
```
Comment (registerrier) [inline gammapy/spectrum/extract.py]: This case requires a test. You could make sure that the correct GTI table is present in the existing tests.

**Comment by JouvinLea** [comment]
@registerrier 
I added the method stacked_gti in the 'SpectrumDatasetOnOffStacker' and I added the test in test_dataset.
I also added a test in test_extract.

**Comment by JouvinLea** [comment]
@adonath we can merge?

**Comment by adonath** [comment]
Thanks @JouvinLea! I did some minor clean-up in a follow-up commit.

**Pull request merged**